src/parser.rs

```diff
@@ -158,7 +158,7 @@ pub type ParseResult<T> = Result<T, ParseError>;
 
 /// Parse takes a sequence of bytes of DER encoded ASN.1 data, constructs a
 /// parser, and invokes a callback to read elements from the ASN.1 parser.
-pub fn parse<'a, T, E: From<ParseError>, F: Fn(&mut Parser<'a>) -> Result<T, E>>(
+pub fn parse<'a, T, E: From<ParseError>, F: FnOnce(&mut Parser<'a>) -> Result<T, E>>(
```

Comment:
For my own edification: why does this have to become `FnOnce`?

Reply:
a) It probably should always have been, we only call it once!
b) It was needed to make the `parse()` call in there work, because of how data is captured.